### PR TITLE
feature/#CU-b4mk8t-multi-datasource-search---Design-provider-pcc-pair-in-request-payload

### PR DESCRIPTION
**Pull request by niloufarmp**
## Changes
Credential object is added to search request and response.

### Search request:
Credentials will be passed as an array of objects.
image

### Search response:
Credentials will be returned for each segment.
image

### CODE REVIEW

**Comment by trip-ninja-inc** [comment]
Task linked: [CU-b4mk8t multi-datasource search - Design provider-pcc pair in request payload](https://app.clickup.com/t/b4mk8t)

**Comment by mcclymont-k** [comment]
Im guess that (on MCFS) each user will have multiple user_creds. Are we going to use the PCC to reference the user Creds we want to access? For amadeus we need the amadeus WSAP and SOAP action but they will all be in the user creds....

**Comment by andrescollart** [comment]
Yeah that makes sense to me @mcclymont-k that we'll use PCC to reference user_creds (including some validation to ensure they're accessing only their own). 

Overall looks good.

**Review by andrescollart — APPROVED** [review]

File: tn_api_spec.yml [hunk 1]
```diff
@@ -1279,16 +1285,40 @@ components:
                                             type: integer
                                             description: Flight duration in minutes
                                             example: 475
+    CredentialInfo:
+        type: object
+        description: Indicate data source to call and required credentials
+        properties:
+            data_source:
```
Comment (Iceman997) [inline tn_api_spec.yml:1292]: Do we need data_source or could we map that from provider?

Comment (Iceman997) [reply tn_api_spec.yml:1292]: Oh I guess some data sources like tripstack etc might not have providers

**Review by Iceman997 — APPROVED** [review]

**Pull request merged**